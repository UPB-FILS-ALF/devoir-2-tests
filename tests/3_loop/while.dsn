& WHILE

SET: %var 200
SET: %r 200
SET: %g 100
SET: %b 200

POSITION: 200 100 20

WHILE: %var
COLOR:  pen %r %g %b
COLOR:  fill %r %g %b
ELLIPSE: 100 40 200 200 
SUB: %r 20
ADD: %g 25
SUB: %b 10
IDIV: %var 20
REPEAT