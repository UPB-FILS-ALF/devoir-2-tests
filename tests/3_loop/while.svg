<?xml version="1.0" encoding="UTF-8" ?>
<svg xmlns="http://www.w3.org/2000/svg" version="1.1" width="1024" height="768">
	<!-- var = 200 -->
	<!-- r = 200 -->
	<!-- g = 100 -->
	<!-- b = 200 -->
	<!-- while var: 200 -->
	<!-- while #0 var: 200 -->
		<!-- pen 200, 100, 200 -->
		<!-- fill 200, 100, 200 -->
		<ellipse cx="100" cy="40" rx="200" ry="200" stroke="rgb(200,100,200)"  stroke-width="1" fill="rgb(200,100,200)"/>
		<!-- r = r - 20 : 180 -->
		<!-- g = g + 25 : 125 -->
		<!-- b = b - 10 : 190 -->
		<!-- var = var / 20 : 10 -->
		<!-- repeat -->
	<!-- while #1 var: 10 -->
		<!-- pen 180, 125, 190 -->
		<!-- fill 180, 125, 190 -->
		<ellipse cx="100" cy="40" rx="200" ry="200" stroke="rgb(180,125,190)"  stroke-width="1" fill="rgb(180,125,190)"/>
		<!-- r = r - 20 : 160 -->
		<!-- g = g + 25 : 150 -->
		<!-- b = b - 10 : 180 -->
		<!-- var = var / 20 : 0 -->
		<!-- repeat -->
</svg>
