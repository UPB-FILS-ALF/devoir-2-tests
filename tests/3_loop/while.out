ERROR LINE (8): POSITION has 2 parameters, you wrote 3
